include/devicetree.h

```diff
@@ -898,7 +924,9 @@
  * @return 1 if the node identifier refers to a usable node,
  *         0 otherwise.
  */
-#define DT_HAS_NODE_STATUS_OKAY(node_id) IS_ENABLED(DT_CAT(node_id, _EXISTS))
+#define DT_HAS_NODE_STATUS_OKAY(node_id) \
+	UTIL_AND(IS_ENABLED(DT_CAT(node_id, _EXISTS)), \
```

Comment:
It's impossible to have a node that has status okay but doesn't exist as far as I can tell. That's why #25009 doesn't have this check. Am I missing something?

Reply:
Nope, you're correct we don't need the extra check.

Reply:
Changed.